atomica/optimization.py

```diff
@@ -1512,8 +1524,11 @@ def constrain_sum_bounded(x: np.array, s: float, lb: np.array, ub: np.array) ->
     :return: Array same size as `x`, such that sum(x)==s and x[i]>=lb and x[i]<=ub for i<len(x)
     :raises: FailedConstraint() if it was not possible to constrain
     """
+    tolerance = 1e-6
 
     # Normalize values
+    if x.sum() == 0:
+        return np.array([s/len(x) for _ in x]) #if all the start values are zero but there is supposed to be a finite value then divide evenly
```

Comment:
@Rowanmh It looks like it's possible on this pathway to bypass the upper/lower constraints if the evenly divided value falls outside a permitted range - would it work to instead skip rescaling `x` if if the sum is 0 but otherwise continue with the algorithm?